### PR TITLE
Improve PHPDoc for package requires

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Replaces #4508.

### CODE REVIEW

File: src/Composer/Package/AliasPackage.php [hunk 1]
```diff
@@ -27,7 +27,9 @@ class AliasPackage extends BasePackage implements CompletePackageInterface
     protected $rootPackageAlias = false;
     protected $stability;
 
+    /** @var Link[] */
     protected $requires;
+    /** @var Link[] */
```
Comment (stof) [inline src/Composer/Package/AliasPackage.php]: should be added for conflicts, provides and replaces too

File: src/Composer/Package/Package.php [hunk 2]
```diff
@@ -441,7 +443,7 @@ public function getReplaces()
     /**
      * Set the recommended packages
      *
-     * @param array $devRequires A set of package links
+     * @param Link[] $devRequires A set of package links
```
Comment (stof) [inline src/Composer/Package/Package.php]: should be be done for other Link list properties too

**Comment by alcohol** [comment]
I agree with @stof, if you're opening a PR for adding docblocks, it would be highly preferable if you add/fixup all relevant docblocks in the files you touch :-)

**Comment by Soullivaneuh** [comment]
@stof @alcohol done. :+1:

**Comment by alcohol** [comment]
Thanks

**Pull request merged**